### DISCUSSION TITLE
Rpm information after parsing process

### DISCUSSION THREAD

**Discussion by huakim** [rpm-software-management/rpm#3484] [General]
Could I get some information stored somewhere under %{_builddir}/%{name}-%{version}-parseinfo after the process of parsing the .spec file has finished and started building, preparing or testing? I think this information
I think this information could be useful, for example, if I need to explicitly specify files that need to be generated. It's okay, if that information will be stored as binary format

**Comment by ghost** [comment]
Maybe, it is better to implement an reflection api for rpmlua? Suppose, i have such rpm spec:
```
Name: binary
Version: 0
Release: 0
Summary: Example
License: None
Requires: second_binary

%description
%{summary}.

%files 

%{lua:
local pkg = rpm.get_package('binary')
local pkgdesc = rpm.get_description("binary")
local summary = pkg.summary
function func(i)
 print("%package -n "..i.name)
 print("Summary: "..summary)
 print("%description -n "..i.name)
 print(pkgdesc)
 print("%files -n "..i.name)
end

for i in ipairs(pkg.requires)
do
  func(i)
end
}
```
This also will be nice, if adding or deleting to the `pkg.requires` or `pkg.provides` and so on will change the state of building package
Also, thanks for metamethods introduced in lua, we can keep pkg table readonly, for example, and make arrays such as requires and provides read-write
There are how readonly tables created:
https://www.lua.org/pil/13.4.5.html
Also, the rpmlua lacks of functionality when working with macros. It is would be nice, if we can see, which macros is already defined, by getting table, consisting all of the names of  macros
